src/java/org/apache/cassandra/db/rows/BufferCell.java

```diff
@@ -43,7 +45,7 @@ public BufferCell(ColumnMetadata column, long timestamp, int ttl, int localDelet
     {
         super(column);
         assert !column.isPrimaryKeyColumn();
-        assert column.isComplex() == (path != null);
+        assert column.isComplex() == (path != null) : format("Column %s.%s(%s: %s) isComplex: %b with cellpath: %s", column.ksName, column.cfName, column.name, column.type.toString(), column.isComplex(), path);
```

Comment:
this is a nit from me; asserts should have messages to help (not needed for the patch, but is useful)